### ISSUE TITLE
Minor doc issue for offset layout

### ISSUE THREAD

**Issue by jonesholger** [LLNL/RAJA#634]
In the documentation find
Offset Layout

```
The RAJA::make_offset_layout method creates a RAJA::Layout object with offsets applied to the indices. For example,:

double* C = new double[11];

RAJA::Layout<1> layout = RAJA::make_offset_layout<1>({{-5}}, {{5}});

RAJA::View<double, RAJA::Layout<1> > Cview(C, layout);

----------------------------------------------------------------------------------

Inject the following into examples/tut_offset-layout.cpp

  const int DIM = 2;

  RAJA::OffsetLayout<DIM> layout =
      RAJA::make_offset_layout<DIM>({{-1, -1}}, {{N_r, N_c}});

  RAJA::Layout<DIM> layout2 =
      RAJA::make_offset_layout<DIM>({{-1, -1}}, {{N_r, N_c}});

```
------------------------------------------------------------------------------
Doesn't compile

**Comment by rhornung67** [comment]
Indeed the documentation is incorrect. I will fix it. 

The issue is that OffsetLayout is a type distinct from Layout (it HAS a Layout and offset data). "Permuted Layout" IS a Layout type. All Layouts have a permutation, so the implementation does not distinguish "Permuted Layout" from Layout. I believe it was done this way to avoid the overhead of offset computations in the data access operator when they are not needed. You could argue that PermutedLayout could be a distinct type, but the implementation would be the same as Layout.  I'll confer with @ajkunen about the details of this decision.

**Comment by ajkunen** [comment]
@rhornung67 has it 100% correct

**Comment by rhornung67** [comment]
Documentation fixed and will appear in v0.9.0 release documentation out soon.
